### ISSUE TITLE
GB is no longer EU

### ISSUE THREAD

**Issue by Timhbl** [driesvints/vat-calculator#124]
VIES (https://ec.europa.eu/taxation_customs/vies/vatRequest.html) dropped support for GB.

Should it be removed from the vat-calculator?

**Comment by driesvints** [comment]
Yes, although you still need to pay taxes when you reach a certain threshold. 

This will be removed in a future version but for now you can overwrite it in the config file if you want.

**Comment by GrahamCampbell** [comment]
See https://www.gov.uk/guidance/the-vat-rules-if-you-supply-digital-services-to-private-consumers#the-place-of-supply-of-digital-services.

**Comment by driesvints** [comment]
@GrahamCampbell first paragraph on the register page:

> You must register your business for VAT with HM Revenue and Customs (HMRC) if its VAT taxable turnover is more than £85,000.

So yes, if you exceed that number only you need to register. In reality I suspect that whoever is doing that probably has better ways to calculate tax than this package 😅

**Comment by GrahamCampbell** [comment]
That line only applies to UK companies.

**Comment by GrahamCampbell** [comment]
It looks like there is actually no registration threshold. Any non-UK entity must register for VAT with HMRC when they make there first digital services sale. https://blog.taxamo.com/insights/vat-and-brexit

**Comment by driesvints** [comment]
That doesn't seems to be true from what some accountants in Belgium say. Honestly I don't think anyone knows what the actual rules are at this point 🤷🏻‍♂️

**Comment by jbarmanet** [comment]
Hi, did anyone managed to disable successfully GB tax rate, or at least put this GB vat rate to 0%, with the new Laravel Spark?
@driesvints I published the config file as advised, and wrote 
```
'rules' => [
    'GB' => 0,
],
```
in it, but the United Kingdom Vat is still stuck to the old 20% rate when selecting the Country...

**Comment by driesvints** [comment]
@jbarmanet no idea why it wouldn't work for you in spark sorry.

**Comment by jbarmanet** [comment]
@driesvints the issue is with the class Mpociot\VatCalculator\VatCalculator as it takes an optional $config argument as constructor... And if you don't pass anything into this constructor (that is what Spark does, it just instantiates the class without any configuration argument:` $vatCalculator = new VatCalculator; `), then the overriden config value is not taken in account when Spark makes the call to `$vatCalculator->getTaxRateForLocation(...)` method.

**Comment by jbarmanet** [comment]
Just sharing my solution so far if it can help anyone, I duplicated the VatCalculator class into a custom location of mine, in order to intercept GB country code and return 0 instead of 0.2 after line 630 in the method getTaxRateForLocation, it looks like this:
```
public function getTaxRateForLocation($countryCode, $postalCode = null, $company = false, $type = null)
    {
        if ($company && strtoupper($countryCode) !== strtoupper($this->businessCountryCode)) {
            return 0;
        }
        $taxKey = 'vat_calculator.rules.'.strtoupper($countryCode);
        if (isset($this->config) && $this->config->has($taxKey)) {
            return $this->config->get($taxKey, 0);
        }
        /**** DULL OVERRIDE HERE ****/
        if (strtoupper($countryCode) == 'GB') {
            return 0;
        }
        // Rest of the class.......... 
```



Then in your composer file you must redirect the original namespace to point to your duplicated-modified file, and exclude the original file to avoid a warning about ambiguous class resolution, as is:
```
"autoload": {
        "psr-4": {
            "App\\": "app/",
            "Mpociot\\VatCalculator\\": "app/Overrides/"
        },
        "exclude-from-classmap": ["vendor/mpociot/vat-calculator/src/Mpociot/VatCalculator/VatCalculator.php"]
    },
```
At the end don't forget to run `composer dump-autoload` to apply your changes.

**Comment by driesvints** [comment]
> that is what Spark does, it just instantiates the class without any configuration argument

Thanks for noting me that. I'm checking in on that 👍

**Comment by driesvints** [comment]
@jbarmanet we released Spark v1.1.8 which fixes that : )
